mitosheet/src/components/LoadingIndicator.tsx

```diff
@@ -1,43 +1,275 @@
 // Copyright (c) Mito
 
 import React, { useEffect, useState } from 'react';
-import LoadingDots from './elements/LoadingDots';
 
 // import css
 import "../../css/loading-indicator.css";
+import { StepType, UpdateType } from '../types';
+import { classNames } from '../utils/classNames';
+import LoadingCircle from './icons/LoadingCircle';
+import NonLoadingCircle from './icons/NonLoadingCircle';
+import { getIcon } from './taskpanes/Steps/StepDataElement';
+
+const isStepEvent = (messageType: string): boolean => {
+    return messageType.endsWith('_edit')
+}
+const isUpdateEvent = (messageType: string): boolean => {
+    return Object.values(UpdateType).includes(messageType as any)
+}
+
+const getDisplayMessageForMessageType = (messageType: StepType | UpdateType): string => {
+    switch (messageType) {
+        case StepType.Initialize: {
+            return 'Created a Mitosheet'
+        }
+        case StepType.AddColumn: {
+            return 'Adding a column'
+        }
+        case StepType.DeleteColumn: {
+            return 'Deleting a column'
+        }
+        case StepType.RenameColumn: {
+            return 'Renaming a column'
+        }
+        case StepType.ReorderColumn: {
+            return 'Reording a column'
+        }
+        case StepType.FillNa: {
+            return 'Filling NaN values'
+        }
+        case StepType.FilterColumn: {
+            return 'Filtering a column'
+        }
+        case StepType.SetColumnFormula: {
+            return 'Setting a formula'
+        }
+        case StepType.DataframeDelete: {
+            return 'Deleting a dataframe'
+        }
+        case StepType.DataframeDuplicate: {
+            return 'Duplicating a dataframe'
+        }
+        case StepType.DataframeRename: {
+            return 'Renaming a dataframe'
+        }
+        case StepType.SimpleImport: {
+            return 'Importing a CSV'
+        }
+        case StepType.Sort: {
+            return 'Sorting a dataframe'
+        }
+        case StepType.Pivot: {
+            return 'Pivoting a dataframe'
+        }
+        case StepType.Merge: {
+            return 'Merging dataframes'
+        }
+        case StepType.Concat: {
+            return 'Concatenating dataframes'
+        }
+        case StepType.DropDuplicates: {
+            return 'Dropping duplicates'
+        }
+        case StepType.ChangeColumnDtype: {
+            return 'Changing a dtype'
+        }
+        case StepType.ChangeColumnFormat: {
+            return 'Formatting'
+        }
+        case StepType.SetCellValue: {
+            return 'Setting a cell value'
+        }
+        case StepType.BulkOldRename: {
+            return 'Bulk old rename'
+        }
+        case StepType.ExcelImport: {
+            return 'Import an Excel file'
```

Comment:
This should be "importing"

<img width="298" alt="Screen Shot 2022-06-20 at 11 19 33 PM" src="https://user-images.githubusercontent.com/18709905/174709247-e02e8b36-3e43-4abc-9905-9b6fa8347fd3.png">